### ISSUE TITLE
Stripe cvc and zip code validation

### ISSUE THREAD

**Issue by ScottDowne** [mozilla/donate.mozilla.org#449]
There is an option in Stripe to decline or accept failed cvc or zip codes.

It is optional, so we can decide we don't need these inputs or decide we do.

I see them increasing our fraud but also our conversion rate, so need to test what works best.

**Comment by ScottDowne** [comment]
In here: https://support.stripe.com/questions/avoiding-fraud-and-disputes

Stripe mentions "Edit account settings to automatically decline charges that fail either AVS or CVC verification. However, even good charges may fail one or more of these checks—investigate which transactions will be blocked before turning on these settings. (These settings aren’t enabled by default and you’ll need to turn them on - you may do so here https://dashboard.stripe.com/account.)"

**Comment by jbuck** [comment]
We already capture the CVC and zip code in our form, so wouldn't it be best
to turn on those checks?

**Comment by ScottDowne** [comment]
@jbuck that's the likely option (we need the stripe account to do this) that's the easy first step.

I think the next step would be to actually question (and test) if we need one or both?

**Comment by valianttry** [comment]
Thought our friend @LRSaunders might have thoughts!

**Comment by LRSaunders** [comment]
"There is an option in Stripe to decline or accept failed cvc or zip codes." - You should definitely turn on that option. That said ensure you are tracking the decline rate against your normal decline rate. It the rate is noticeably higher, you will need to adjust.   
I would also only ask for CVV and zip instead of the entire address (not sure what you decided to do there)

**Comment by ScottDowne** [comment]
@LRSaunders Yeah, I'm not sure about the historical reasoning for the entire address, but it's a good question, it likely has no good reason to be there anymore. Filing that.

**Comment by valianttry** [comment]
I am absolutely in favor or eliminating full address for donation. However, we may want to ask for (if not require) postal code and / or country at minimum for email optins.

**Comment by adamlofting** [comment]
Just linking this ticket to the related one about testing this #163

**Comment by ScottDowne** [comment]
I wonder if asking for country is needed. I'm thinking just postal code (for credit card reasons).

If we want country for signup reasons, we could. We should test this though so we know what effect it has on our conversion. For whatever it's worth, singup does store locale.

**Comment by ScottDowne** [comment]
Actually, we should do this in steps. First remove everything but postal code and country, that's already better than what we have.

Then we may want to run a test without country.

**Comment by jbuck** [comment]
I think you need the country to validate the postal code, right?

**Comment by ScottDowne** [comment]
@jbuck maybe, that's a good question, but I was under the impression that cvc is all you really need. See the stripe checkout widget. I think the rest is additional fraud checks against the credit card, and not against the address.

So stripe would check the postal code against the credit card, and not validate it in and of itself?

**Comment by valianttry** [comment]
@LRSaunders might know better, but I believe all Stripe does is validate that the zip code entered matches the one on file with the credit card company. I want to double check that we DO have a version of the form that only asks for first name, last name, CC#, mm/yy, CVC and zip? (e.g. no street, city, or country).

**Comment by LRSaunders** [comment]
Stripe has a couple different options. They validate the address if you add it ( I think they just use the first numbers) They also validate with just the zip. 

All you technically need is cvv but zip is recommended for cases of fraud.

**Comment by ScottDowne** [comment]
@valianttry We do have a similar form for what you're asking, but it still asks for country. So first name, last name, CC#, mm/yy, CVC, country and zip

https://donate.mofostaging.net/en-US/?test=address-test

Initially the comp had country and no zip. I pushed a bit back on that going for zip and not country. I ended up with both because it was still better than everything.

I think there is also a slight bonus to getting country for signup reasons, but it's optional.

I'm happy to test all this, as it's configurable, but I'm also happy to just have someone make a decision and we just do it.
